### ISSUE TITLE
[PHP] Heap Corruption when keep_descriptor_pool_after_request is active

### ISSUE THREAD

**Issue by simonberger** [protocolbuffers/protobuf#16894]
When protobuf messages are created in PHP scripts executed in CLI, `zend_mm_heap corrupted` errors and memory leaks are reported.

I created a reproducer repository here. 
https://github.com/simonberger/protobuf-reproducer

The only thing needed is initializing any protobuf message. I picked the available`Google\Protobuf\Api` message. With codeception the output is

```
Tests.Protobuf Tests (1) ------------------------------------------------------------------------------------------------------------------------------------------------------------
✔ ProtobufTest: Protobuf(0.01s)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Time: 00:00.110, Memory: 10.00 MB

OK (1 test, 0 assertions)
zend_mm_heap corrupted
```

PHP and Extension versions:
- The docker image is using newest debian with PHP 8.3.7 and protobuf extension 3.25.3
- I reproduced it also on a Mac M1 with PHP 8.3.7 and protobuf extension 4.26.1

I started the repository with executing it in a codeception test, because this is what we are using.
The problem can be reproduced much easier. A call to a PHP script just with the message creation is enough.
The script is also added to the repository.

```php
<?php
use Google\Protobuf\Api;

$message = new Api();
// Not necessary for the zend_mm_heap corrupted error to appear.
$message->setName('TestApi');
```
 `php public/protobuf_init.php`

These problems where already reported in the old bug https://github.com/protocolbuffers/protobuf/issues/9215 and not only in commands. We do not yet use the protobuf extension in production and just noticed the errors while executing tests.

**Comment by simonberger** [comment]
Problems with the option aren't CLI only.
After we ran first attempts on tests servers, we noticed the following errors with option enabled.
I have no details at this point and also hope they will be solved with a fix for the CLI problems as well.

- Error appears: `Couldn't find descriptor. Note only generated code may derive from \Google\Protobuf\Internal\Message` This happens in a PHP 8.2 project only, not in a 8.3 based one. It happens with a high chance, but not always. Possibly when more requests access the data at the same time
- segmentation faults randomly here and there not yet matched to specific requests but starting with the deployment of the protobuf extension and seemingly ending after disabling the option but keeping the extension active.

We use newest PHP 8.2 and 8.3 versions in apache with mod_php.
